draft-ietf-lamps-csr-attestation.md

```diff
@@ -523,16 +551,120 @@ ext-evidence EXTENSION ::= {
+
+This list is left unconstrained in this document. However, implementers can
+populate it with the formats that they wish to support.
+
+~~~
+AttestationResult ::= SEQUENCE {
+   type   ATTESTATION-RESULT.&id({AttestationResultSet}),
+   stmt   ATTESTATION-RESULT.&Type({AttestationResultSet}{@type}),
+}
+~~~
+{: #code-AttestationResult title="Definition of AttestationResult"}
+
+In {{code-AttestationResult}}, type is an OID that indicates the format of the
+value of stmt.
+
+~~~
+AttestationResultBundle ::= SEQUENCE SIZE (1..MAX) OF AttestationResult
+~~~
+
+~~~
+id-aa-ar OBJECT IDENTIFIER ::= { id-ata 60 }
+
+-- For PKCS#10
+attr-ar ATTRIBUTE ::= {
+  TYPE AttestationResultBundle
+  COUNTS MAX 1
+  IDENTIFIED BY id-aa-ar
+}
+
+-- For CRMF
+ext-ar EXTENSION ::= {
+  SYNTAX AttestationResultBundle
+  IDENTIFIED BY id-aa-ar
+}
+~~~
+{: #code-extensions-ar title="Definitions of CSR attribute and extension"}
+
+## Implementation Considerations
+
+This specification is applicable both in cases where a CSR
+is constructed internally or externally to the Attesting Environment, from the
+point of view of the calling application. This section is particularly
+applicable to the background check model.
+
+Cases where the CSR is generated internally to the Attesting Environment
+are straightforward: the Hardware Security Model (HSM) generates and embeds
+the Evidence and the corresponding certification paths when constructing the CSR.
+
+Cases where the CSR is generated externally may require extra communication
+between the CSR generator and the Attesting Environment, first to obtain
+the necessary Evidence about the subject key, and then to use
+the subject key to sign the CSR; for example, a CSR generated by
+a popular crypto library about a subject key stored on a PKCS#11 {{PKCS11}} device.
+
+As an example, assuming that the HSM is, or contains, the Attesting Environment and
+some cryptographic library is assembling a CSR by interacting with the HSM over some
+network protocol, then the interaction would conceptually be:
+
+~~~ aasvg
+                   +---------+          +-----+
+                   | Crypto  |          | HSM |
+                   | Library |          |     |
+                   +---------+          +-----+
+                        |                  |
+                        | getEvidence()    |
+                        |----------------->|
+                        |                  |
+                        |<-----------------|
++---------------------+ |                  |
+| CSR = assembleCSR() |-|                  |
++---------------------+ |                  |
+                        |                  |
+                        | sign(CSR)        |
+                        |----------------->|
+                        |                  |
+                        |<-----------------|
+                        |                  |
+~~~
+{: #fig-csr-client-p11 title="Example interaction between CSR generator and HSM."}
```

Comment:
I think its confusing for a crypto lib to have getEvidence calls. This dissonance is easier to see if you also have a diagram for getAttestationResultToken(). Both the AR token and Evidence are already signed (by Verifier / Attesting Env respectively) so there isn't a crypto operation needed to fetch them.